lib/pages/home/home.dart

```diff
@@ -126,10 +130,16 @@ class OBHomePageState extends ReceiveShareState<OBHomePage>
   }
 
   @override
-  void onShare(Share share) {
+  void onShare(Share share) async {
     debugPrint("received share to file " + share.path);
-    _modalService.openCreatePost(
-        context: context, image: File.fromUri(Uri.parse(share.path)));
+    var image = File.fromUri(Uri.parse(share.path));
+    if (!await _validationService.isImageAllowedSize(image, OBImageType.post)) {
+      int limit = _validationService.getAllowedImageSize(OBImageType.post);
+      _toastService.error(
```

Comment:
I think the int limit has to be divided for it to be MBs, so  `limit = limit ~/ 1048576;` 🙈

Reply:
Oh.. Yeah, copied that from the photo picker, but seems like I missed that...